client/src/app/game/chat/chat.tsx

```diff
@@ -123,104 +149,179 @@ function Chat() {
 	}
 
 	return (
-		<div
-			className={css.chat}
+		<ChatContent
 			style={style}
-			onClick={(e) => {
-				dispatch({
-					type: localMessages.SETTINGS_SET,
-					setting: {
-						key: 'chatSize',
-						value: {
-							w: e.currentTarget.offsetWidth,
-							h: e.currentTarget.offsetHeight,
-						},
-					},
-				})
-			}}
-		>
-			<div className={css.header} {...bindChatPos()}>
-				<p>Chatting with {opponentName}</p>
-				<Button
-					onClick={() =>
-						dispatch({
-							type: localMessages.SETTINGS_SET,
-							setting: {
-								key: 'showBattleLogs',
-								value: !showLog,
-							},
+			onClick={resizeChat}
+			chatMessages={chatMessages.map((line) => {
+				let isOpponent: boolean
+				if (isSpectator) {
+					isOpponent =
+						!!players &&
+						[order[1], players[order[1]]?.playerId].includes(line.sender.id)
+				} else {
+					isOpponent = ![playerId, playerEntity].includes(line.sender.id)
+				}
+
+				let sender: 'playerOne' | 'playerTwo' | 'spectator' = isOpponent
+					? 'playerTwo'
+					: 'playerOne'
+
+				if (
+					!!players &&
+					![
+						order[1],
+						players[order[1]]?.playerId,
+						order[0],
+						players[order[0]]?.playerId,
+					].includes(line.sender.id)
+				) {
+					sender = 'spectator'
+				}
+
+				return {
+					message: line.message,
+					sender,
+					createdAt: line.createdAt,
+					isBattleLogMessage: line.sender.type === 'system',
+				}
+			})}
+			showLog={showLog}
+			profanityFilterEnabled={settings.profanityFilterEnabled}
+			isSpectating={isSpectator}
+			playerNames={[playerName, opponentName]}
```

Comment:
`playerName` can be the name of the player as a spectator, if Beef is spectating p2 it shows "BEEF'S TURN" instead of "P2'S TURN"

Reply:
Fixed it!